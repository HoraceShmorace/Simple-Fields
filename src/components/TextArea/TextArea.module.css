textarea:invalid {
  border-color: red;
}

textarea:valid {
  border-color: black;
}
.root textarea {
  background-color: #f5f5f5;
  border: 1px solid transparent;
  border-radius: 8px;
  color: #444;
  font-family: helvetica;
  font-size: 1rem;
  font-weight: bold;
  height: 4rem;
  margin-left: -2px;
  outline-width: 0;
  padding: 6px;
  width: calc(100% - 12px);
}

.root textarea::placeholder {
  color: #aaa;
  font-size: 1.5rem;
}

