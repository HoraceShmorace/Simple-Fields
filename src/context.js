import { createContext, useReducer } from 'react'
import { node } from 'prop-types'
const ADD_DATA_ACTION = 'ADD_DATA'
const RESET_DATA_ACTION = 'RESET_DATA'
const initialState = {}
const reducer = (state, { payload, type }) => {
  switch (type) {
    case ADD_DATA_ACTION:
      return {
        ...state,
        ...payload
      }
    case RESET_DATA_ACTION:
      return {}
    default:
      return state
  }
}

export const Context = createContext(initialState)

export const ContextProvider = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState)
  const addData = payload => dispatch({ payload, type: ADD_DATA_ACTION })
  const resetData = () => dispatch({ payload: initialState, type: RESET_DATA_ACTION })

  return (
    <Context.Provider value={{ addData, resetData, state }}>
      {children}
    </Context.Provider>
  )
}

ContextProvider.propTypes = {
  children: node.isRequired
}
