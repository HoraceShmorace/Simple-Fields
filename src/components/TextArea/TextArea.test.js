import { render } from 'enzyme'
import toJson from 'enzyme-to-json'
import TextArea from './TextArea'
import { ContextProvider } from '~src/context'
import mockData from '~mock/form.json'

const FIELD_TYPE_TEXT = 'textarea'
const { fields, id } = mockData

describe('TextArea', () => {
  describe('when rendered', () => {
    const props = fields.find(({ type }) => type === FIELD_TYPE_TEXT)
    const component = render(
      <ContextProvider formId={id}>
        <TextArea {...props} />
      </ContextProvider>
    )

    it('should match snapshot', () => {
      expect(toJson(component)).toMatchSnapshot()
    })
  })
})
