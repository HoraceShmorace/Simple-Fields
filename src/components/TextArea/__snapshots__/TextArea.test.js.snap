// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea when rendered should match snapshot 1`] = `
<textarea
  aria-required="false"
  id="115390398"
  maxlength="126"
  name="115390398"
  placeholder=""
/>
`;
